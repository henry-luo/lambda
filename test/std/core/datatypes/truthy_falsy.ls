// Test: Truthy and Falsy Values
// Category: core/datatypes
// Type: positive
// Expected: 1

// Falsy values
let f1 = !!false      
f1
let f2 = !!0          
f2
let f3 = !!""         
f3
let f4 = !!null       
f4
let f5 = !!undefined  
f5
let f6 = !!NaN        
f6
let t1 = !!true           
t1
let t2 = !!1              
t2
let t3 = !!-1             
t3
let t4 = !!"hello"        
t4
let t5 = !![]            
t5
let t6 = !!{}            
t6
let t7 = !![0]           
t7
let t8 = !!function(){}  
t8
let result1 = "" || "default"     
result1
let result2 = 0 || 42            
result2
let result3 = null ?? "fallback"  
result3
1
