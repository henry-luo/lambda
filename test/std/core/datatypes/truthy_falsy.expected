09:15:42 [NOTICE] Start transpiling test/std/core/datatypes/truthy_falsy.ls...
09:15:42 [ERROR] Syntax tree has errors.
09:15:42 [ERROR] Syntax error at Ln 7, Col 10 - 7, Col 12: ERROR
09:15:42 [ERROR] ERROR node at Ln 7, Col 10 - 7, Col 12
09:15:42 [ERROR] Syntax error at Ln 8, Col 10 - 8, Col 12: ERROR
09:15:42 [ERROR] ERROR node at Ln 8, Col 10 - 8, Col 12
09:15:42 [ERROR] Syntax error at Ln 9, Col 10 - 18, Col 18: ERROR
09:15:42 [ERROR] ERROR node at Ln 9, Col 10 - 18, Col 18
09:15:42 [ERROR] Syntax error at Ln 25, Col 22 - 25, Col 29: ERROR
09:15:42 [ERROR] ERROR node at Ln 25, Col 22 - 25, Col 29
09:15:42 [ERROR] Syntax error at Ln 27, Col 24 - 27, Col 32: ERROR
09:15:42 [ERROR] ERROR node at Ln 27, Col 24 - 27, Col 32
09:15:42 [ERROR] Missing node at Ln 31, Col 1: expected "
09:15:42 [ERROR] Error: Failed to compile the function.
##### Script 'test/std/core/datatypes/truthy_falsy.ls' executed: #####
null
